src/main/java/calculator/CalculatorService.java

```diff
@@ -0,0 +1,132 @@
+
+    @Override
+    public int calculate(String ID) {
+        String expression = getFinalExpression(ID);
+        MathParser mathParser = MathParserFactory.create();
+
+        return mathParser.calculate(expression)
+                .doubleValue()
+                .intValue();
+    }
+
+    @Override
+    public int addData(String ID, String parameterName, String paramValue) {
+        String valueFromRepository = getValueFromRepository(ID, parameterName);
+        if (!isDataExists(ID)) {
+            REPOSITORY.putNewData(ID);
+        }
+        int statusCode = getStatusCode(valueFromRepository, parameterName, paramValue);
+        if (!isBadFormatExpression(paramValue, parameterName) && !isParameterHasOverLimitValue(paramValue)) {
+            REPOSITORY.update(ID, parameterName, paramValue);
+        }
+        return statusCode;
+    }
+
+    @Override
+    public void deleteData(String ID, String parameterName) {
+        Map<String, String> data = REPOSITORY.getDataByID(ID);
+
+        data.remove(parameterName);
+    }
+
+    private String getFinalExpression(String ID) {
+        List<String> expression = getExpression(ID);
+        Map<String, String> repositoryData = REPOSITORY.getDataByID(ID);
+
+        while (isExpressionWithVariables(repositoryData, expression)) {
+            convertExpressionWithValue(expression, ID);
+        }
+
+        return deleteSpacesAndConvertListToString(expression);
+    }
+
+    private List<String> getExpression(String ID) {
+        Map<String, String> data = REPOSITORY.getDataByID(ID);
+
+        return Arrays.asList(data
+                .get(EXPRESSION)
+                .split(EMPTY_SYMBOL));
+    }
+
+    private boolean isDataExists(String ID) {
+        return REPOSITORY.getRepositoryData().containsKey(ID);
+    }
+
+    private void convertExpressionWithValue(List<String> expression, String ID) {
+        Map<String, String> data = REPOSITORY.getDataByID(ID);
+
+        for (int i = 0; i < expression.size(); i++) {
+            String key = expression.get(i);
+            if (data.containsKey(key)) {
+                expression.set(i, data.get(key));
+            }
+        }
+    }
+
+    private boolean isExpressionWithVariables(Map map, List<String> expression) {
+        return expression.stream().anyMatch(map::containsKey);
+    }
+
+    private String getValueFromRepository(String ID, String parameterName) {
+        Map<String, String> data = REPOSITORY.getDataByID(ID);
+        if (data == null) {
+            return null;
+        } else {
+            return data.get(parameterName);
+        }
+    }
+
+    private boolean isBadFormatExpression(String paramValue, String parameterName) {
+        return parameterName.equals(EXPRESSION) &&
+                !(paramValue.contains(PLUS) || paramValue.contains(MINUS) ||
+                        paramValue.contains(DIVIDE) || paramValue.contains(MULTIPLY));
```

Comment:
1. It would be better to format code with starting && or || symbols
for example
```
return paramValue.contains(PLUS)
                || paramValue.contains(MINUS)
                || paramValue.contains(DIVIDE)
                || paramValue.contains(MULTIPLY);
```
2. Code can be moved to separate method. For example "isSpecialSymbol(....)"
```
paramValue.contains(PLUS) || paramValue.contains(MINUS) ||
                        paramValue.contains(DIVIDE) || paramValue.contains(MULTIPLY)
```